docs/DesignDocs/SGX_EMM.md

```diff
@@ -0,0 +1,678 @@
+ *                     order, address preference, page type. The untrusted side
+ *    implementation should always invoke mmap syscall with MAP_SHARED|MAP_FIXED_NOREPLACE, and
+ *    translate following additional bits to proper parameters invoking mmap or other SGX specific
+ *    syscall(s) provided by the kernel.
+ *        The flags param of this interface should include exactly one of following for committing mode:
+ *            - SGX_EMA_RESERVE: kernel map an address range with PROT_NONE, no EPC EAUGed.
+ *            - SGX_EMA_COMMIT_NOW: reserves memory range with SGX_EMA_PROT_READ|SGX_EMA_PROT_WRITE, if supported,
+ *                   kernel is given a hint to EAUG EPC pages for the area as soon as possible.
+ *            - SGX_EMA_COMMIT_ON_DEMAND: reserves memory range, EPC pages can be EAUGed upon #PF.
+ *        ORed with zero or one of the committing order flags:
+ *            - SGX_EMA_GROWSDOWN: if supported, a hint given for the kernel to EAUG pages from higher
+ *                              to lower addresses, no gaps in addresses above the last committed.
+ *            - SGX_EMA_GROWSUP: if supported, a hint given for the kernel to EAUG pages from lower
+ *                              to higher addresses, no gaps in addresses below the last committed.
+ *        Optionally ORed with
+ *            -  SGX_EMA_ALIGNED(n):	Align the region on a requested	boundary.
+ *                           Fail if a suitable region cannot be found,
+ *                           The argument n specifies the binary logarithm of
+ *                           the desired alignmentand must be at least 12.
+ *        Optionally ORed with one of following page types:
+ *             - SGX_EMA_PAGE_TYPE_REG: regular page type. This is the default if not specified.
+ *             - SGX_EMA_PAGE_TYPE_SS_FIRST: the first page in shadow stack.
+ *             - SGX_EMA_PAGE_TYPE_SS_REST: the rest page in shadow stack.
+ * @retval 0 The operation was successful.
+ * @retval EINVAL Any parameter passed in is not valid.
+ * @retval errno Error as reported by dependent syscalls, e.g., mmap().
+ */
+int sgx_mmap_ocall(void *addr, size_t length, int prot, int flags);
+
+/*
+ * Call OS to change permissions, type, or notify EACCEPT done after TRIM
+ *
+ * @param[in] addr start address of the memory to change protections.
+ * @param[in] length length of the area.  This must be a multiple of the page size.
+ * @param[in] flags this must be OR'ed of following:
+ *            SGX_EMA_PROT_READ
+ *            SGX_EMA_PROT_WRITE
+ *            SGX_EMA_PROT_EXEC
+ *            SGX_EMA_PAGE_TYPE_TRIM: change the page type to PT_TRIM, implies RW.
+ *            SGX_EMA_PAGE_TYPE_TCS: change the page type to PT_TCS
+ *            SGX_EMA_PROT_NOACCESS: Signal the kernel EACCEPT is done for PT_TRIM pages.
+ * @retval 0 The operation was successful.
+ * @retval EINVAL if any parameter passed in is not valid.
```

Comment:
```suggestion
 * @retval EINVAL A parameter passed in is not valid.
```
retval generates a table where everything after the first token is in a separate column.  Use complete sentences line lines 500 and 502 do, no "if"